### PR TITLE
Added navigation from main to detail screens

### PR DESCRIPTION
**Pull request by Blazevarjo**
- Added navigation from main to detail screens
- Switched bottom navigation to the one supported by react-navigation
- Added context to share pokemon data in the whole app and fetch data only once
- Removed ApiTesting file because it's not needed anymore

### CODE REVIEW

File: src/screens/details/General.js [hunk 1]
```diff
@@ -1,18 +1,17 @@
...
 
 export default function General() {
-  // Just for testing:
-  const example = {
-    sprite:
-      'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/1.png',
-    color: 'green',
-  };
+  const {
+    currentPokemon: { color, sprite },
```
Comment (kamil-matula) [inline src/screens/details/General.js]: This part results in crashing the application :(

File: src/screens/main/HomeTabs.js [hunk 2]
```diff
@@ -1,34 +1,24 @@
 import React, { useState } from 'react';
```
Comment (kamil-matula) [inline src/screens/main/HomeTabs.js]: 'UseState' is not used now, so you can delete it

File: src/screens/main/HomeTabs.js [hunk 3]
```diff
@@ -1,34 +1,24 @@
...
       barStyle={{ backgroundColor: colors.white }}
       inactiveColor='#808080'
-      navigationState={{ index, routes }}
-      onIndexChange={setIndex}
-      renderScene={renderScene}
       shifting={true}
-    />
+    >
+      <Tab.Screen name='Home' component={Home} options={{ tabBarIcon: 'home' }} />
+      <Tab.Screen name='heart' component={Favourites} options={{ tabBarIcon: 'information' }} />
```
Comment (kamil-matula) [inline src/screens/main/HomeTabs.js]: The name should be 'Favourites' and the icon should be 'heart'

**Review by kamil-matula — CHANGES_REQUESTED** [review]

**Review by kamil-matula — APPROVED** [review]
Good job! :)

**Review by piotr-wawro — APPROVED** [review]

**Pull request merged**